### PR TITLE
patch ajv and also make oas async function

### PR DESCRIPTION
**Pull request by supertong**
The `validateInner` is actually an async function so we have to make our `oas` an async function too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { oas } from 'koa-oas3';
 
 const app = new Koa();
 app.use(bodyParser());
-app.use(oas({
+app.use(await oas({
```
Comment (lucaslago) [inline README.md:23]: I'd extract this to a variable as an example, might be better for the compiler

Comment (supertong) [reply README.md:23]: Nah, every request shares the same middleware that's created from this `await oas()` function.

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     "jsonfile": "^5.0.0",
     "koa-bodyparser": "^4.2.1",
     "koa-compose": "^4.1.0",
-    "oas-validator": "^3.3.1",
+    "oas-validator": "lucaslago/oas-kit#oas-validator-v4.0.7-gitpkg",
```
Comment (lucaslago) [inline package.json:33]: :( I hope we can get rid of this soon

File: __tests__/index.spec.ts [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
-import { ChowOptions } from 'oas3-chow-chow';
 
-describe('Koa Oas3', () => {
-  const mw = oas({
-    file: path.resolve('./__tests__/fixtures/pet-store.json'),
-    endpoint: '/openapi',
-    uiEndpoint: '/openapi.html',
-    validatePaths: ['/pets']
-  });
+describe('Koa Oas3', async () => {
```
Comment (lucaslago) [inline __tests__/index.spec.ts:9]: You don't need async here, jest is complaining too: https://travis-ci.org/github/atlassian/koa-oas3/builds/714344155

File: src/index.ts [hunk 4]
```diff
@@ -133,11 +133,14 @@ function loadFromFile(file?: string) {
...
 }
 
-function compileOas(config: Config) {
+async function compileOas(config: Config) {
   let openApiObject: any = config.spec || loadFromFile(config.file);
-  if (!oasValidator.validateSync(openApiObject, {})) {
+  try {
+    await oasValidator.validateInner(openApiObject, {});
+  } catch (err) {
     throw new Error('Invalid Openapi document');
```
Comment (lucaslago) [inline src/index.ts:141]: Do we have a test for this scenario?

Comment (supertong) [reply src/index.ts:144]: Good point, let me check.

Comment (supertong) [reply src/index.ts:144]: Added a test to covered 👍

File: __tests__/index.spec.ts [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
-import { ChowOptions } from 'oas3-chow-chow';
 
-describe('Koa Oas3', () => {
-  const mw = oas({
-    file: path.resolve('./__tests__/fixtures/pet-store.json'),
-    endpoint: '/openapi',
-    uiEndpoint: '/openapi.html',
-    validatePaths: ['/pets']
-  });
+describe.only('Koa Oas3', () => {
```
Comment (lucaslago) [inline __tests__/index.spec.ts:9]: describe.only here, prob want to remove that change

File: __tests__/index.spec.ts [hunk 6]
```diff
@@ -301,43 +318,47 @@ describe('Koa Oas3', () => {
...
+//       body: {
+//         id: 1,
+//         tag: 'tag'
+//       }
+//     });
+//     const next = jest.fn();
+//     return expect(mw(ctx, next)).rejects.toThrow();
+//   })
+
+// })
```
Comment (lucaslago) [inline __tests__/index.spec.ts:364]: Why the comments?

File: __tests__/index.spec.ts [hunk 7]
```diff
@@ -19,6 +19,16 @@ describe.only('Koa Oas3', () => {
...
 
+  test('It should throw error if invalid Openapi document is passed', async () => {
+    return expect(oas({
+      file: path.resolve('./__tests__/fixtures/broken.json'),
+      endpoint: '/openapi',
+      uiEndpoint: '/openapi.html',
+      validatePaths: ['/pets'],
+      validationOptions: { requestBodyAjvOptions: { allErrors: true } } as ChowOptions
+    })).rejects.toThrow('Invalid Openapi document');
+  })
```
Comment (lucaslago) [inline __tests__/index.spec.ts:30]: Nice one!

File: __tests__/index.spec.ts (hunk 6 above)
Comment (supertong) [reply __tests__/index.spec.ts]: Was debugging something. I will put this back.

File: __tests__/index.spec.ts (hunk 5 above)
Comment (supertong) [reply __tests__/index.spec.ts]: Done.

**Pull request merged**